trade-finance-manager-ui/server/api.js

```diff
@@ -1230,6 +1230,35 @@ const addFeesToAnExistingPayment = async (reportId, feeRecordIds, paymentIds, us
   return response.data;
 };
 
+/**
+ * @param {string} dealId - The deal ID
+ * @param {Partial<import('@ukef/dtfs2-common').TfmDealCancellation>} cancellationUpdate - The deal cancellation update object
+ * @param {string} userToken - The user token
+ * @returns {Promise<void>}
+ */
+const updateDealCancellation = async (dealId, cancellationUpdate, userToken) => {
+  try {
+    const isValidDealId = isValidMongoId(dealId);
+
+    if (!isValidDealId) {
+      console.error('updateDealCancellation: Invalid deal id provided %s', dealId);
+      return { status: 400, data: 'Invalid deal id' };
```

Comment:
I assume this is an existing pattern but its weird to return `{ status: 400 }` when we've not even made an api request?

Reply:
Looking at the payment record functions above they dont try/catch or even validate - would that be a better plan - let the api's deal with this stuff?

Reply:
That's a much better idea - I've removed it and let the api handle it. It feels like all the calls in this file should not have the valid id check given they're all checked anyway in the API layer! So a lot of repetition around.